ethereumj-core/src/main/java/org/ethereum/listener/EventListener.java

```diff
@@ -0,0 +1,304 @@
+                        if (!pendings.containsKey(new ByteArrayWrapper(receipt.getTransaction().getHash()))) {
+                            // ask PendingState to track candidate transactions from blocks since there is
+                            // no guarantee the transaction of interest received as a pending
+                            // on included transaction PendingState should call onPendingTransactionUpdateImpl
+                            // with INCLUDED state
+                            // if Tx was included into fork block PendingState should call onPendingTransactionUpdateImpl
+                            // with PENDING state
+                            pendingState.trackTransaction(receipt.getTransaction());
+                        }
+                    }
+                }
+            }
+
+            if (blockSummary.getTotalDifficulty().compareTo(lastTotDiff) > 0) {
+                lastTotDiff = blockSummary.getTotalDifficulty();
+                bestBlock = blockSummary.getBlock();
+                // we need to update pendings bestConfirmingBlock
+                newBestBlock(blockSummary.getBlock());
+            }
+        } catch (Exception e) {
+            logger.error("Unexpected error while processing onBlock", e);
+        }
+    }
+
+    public void onPendingTransactionUpdateImpl(TransactionReceipt txReceipt, PendingTransactionState state, Block block) {
+        try {
+            if (state != PendingTransactionState.DROPPED || pendings.containsKey(new ByteArrayWrapper(txReceipt.getTransaction().getHash()))) {
+                logger.debug("onPendingTransactionUpdate: " + Hex.toHexString(txReceipt.getTransaction().getHash()) + ", " + state);
+            }
+            onReceipt(txReceipt, block, state);
+        } catch (Exception e) {
+            logger.error("Unexpected error while processing onPendingTransactionUpdate", e);
+        }
+    }
+
+    /**
+     *  Initialize listener with the contract of interest
+     */
+    public void setContract(String abi, byte[] contractAddress) {
+        contract = new CallTransaction.Contract(abi);
+        contractAddressFilter = new LogFilter().withContractAddress(contractAddress);
+    }
+
+    // checks the Tx receipt for the contract LogEvents
+    // initiated on [onPendingTransactionUpdateImpl] callback only
+    private synchronized void onReceipt(TransactionReceipt receipt, Block block, PendingTransactionState state) {
+        ByteArrayWrapper txHashW = new ByteArrayWrapper(receipt.getTransaction().getHash());
+        if (contractAddressFilter.matchBloom(receipt.getBloomFilter())) {
+            int txCount = 0; // several transactions are possible withing a single Ethereum Tx
+            List<EventData> matchedTxs = new ArrayList<>();
+            for (LogInfo logInfo : receipt.getLogInfoList()) {
+                if (contractAddressFilter.matchBloom(logInfo.getBloom()) &&
```

Comment:
Current implementation doesn't allow user to instantiate a listener which does topics filtering out of the box. I think it worth to be extended with that feature.

Reply:
Do you think we need such kind of low-level listener in addition to parsed events listener?